cclib/parser/turbomoleparser.py

```diff
@@ -109,19 +112,60 @@ def extract(self, inputfile, line):
             nmo = int(line.split('=')[1])
             self.set_attribute('nbasis', nmo)
             self.set_attribute('nmo', nmo)
-
+            
+        # The DFT functional.
+        # This information is printed by dscf but not in an easily parsible format, so we'll take it from the control file instead...
+        # Additionally, turbomole stores functional names in lower case. This looks odd, so we'll convert to uppercase (?)
+        if line[3:13] == "functional":
+            self.metadata['functional'] = line.split()[1].upper()
+        
+        # The DFT functional.
+        # This information is printed by dscf but not in an easily parsable format, so we'll take it from the control file instead...
+        # Additionally, turbomole stores functional names in lower case. This looks odd, so we'll convert to uppercase (?)
+        if line[3:13] == "functional":
+            # $dft
+            #    functional b-p
+            self.metadata['functional'] = line.split()[1].upper()
+            self.DFT = True
+        
+        if "density functional" in line:
+            self.DFT = True
+        
         # Extract the version number and optionally the build number.
         searchstr = ": TURBOMOLE"
         index = line.find(searchstr)
         if index > -1:
             line = line[index + len(searchstr):]
             tokens = line.split()
-            package_version = tokens[0][1:].replace("-", ".")
+            # This line could look like any of the following:
+            # - TURBOMOLE rev. V7.4.1 (2bfdd732)

```

Comment:
It might be cleaner at this point to extract the `VX.Y.Z` and the revision using regexes.

Reply:
Done, regex is not my forte I'm afraid so there may be a better solution...